sdk/python/feast/infra/offline_stores/bigquery_source.py

```diff
@@ -61,6 +60,14 @@ def __init__(
             table = table_ref
         self.bigquery_options = BigQueryOptions(table_ref=table, query=query)
 
+        if date_partition_column:
+            warnings.warn(
```

Comment:
do we want to add a specific version warning? e.g. Feast 0.21+ onwards will not support

and same for other sources

Reply:
Yeah I'll add that